File: 3° semestre/Tecnicas-de-programacão-2/aula-24-10-11/Proxy/Proxy01.py
# Objeto real
class ServicoReal:
    def executar(self):
        print("Obejto real - executando o servico real")

# Proxy (procurador)
class ProxyServicoReal:
    def __init__(self):
        self.servicoReal = ServicoReal()

    def executar(self):
        print("Proxy - antes da execução do serviço")
        self.servicoReal.executar()
        print("Proxy - depois da execução do serviço")

# uso do proxy
proxy = ProxyServicoReal()
proxy.executar()

File: 3° semestre/Tecnicas-de-programacão-2/aula-24-10-11/Proxy/Proxy01.js
// objeto real
class ServicoReal {
    executar() {
        console.log("Obejto Real - Executando o serviço real.");
    }
}

// Proxy (Procurador)
class ProxyServicoReal {
    constructor() {
        this.ServicoReal = new ServicoReal();
    }

    executar() {
        console.log("Proxy - Antes da execução do serviço");
        this.ServicoReal.executar();
        console.log("Proxy - Depois da execução do serviço");
    }
}

// Uso do proxy:
const proxy = new ProxyServicoReal();
proxy.executar();
